Sources/AuthFoundation/JWT/Internal/DefaultIDTokenValidator.swift

```diff
@@ -18,61 +18,72 @@ import CommonCrypto
         guard let tokenIssuerString = token.issuer,
-              let tokenIssuer = URL(string: tokenIssuerString)
-        else {
-            throw JWTError.invalidIssuer
-        }
-
-        guard tokenIssuer.absoluteString == issuer.absoluteString
+              let tokenIssuer = URL(string: tokenIssuerString),
+              tokenIssuer.absoluteString == issuer.absoluteString
         else {
             throw JWTError.invalidIssuer
         }
-
-        guard token[.audience] == clientId else {
-            throw JWTError.invalidAudience
-        }
-
-        guard tokenIssuer.scheme == "https" else {
-            throw JWTError.issuerRequiresHTTPS
-        }
-        
-        guard token.header.algorithm == .rs256 else {
-            throw JWTError.unsupportedAlgorithm(token.header.algorithm)
-        }
         
-        guard let expirationTime = token.expirationTime,
-              expirationTime > Date.nowCoordinated else {
-            throw JWTError.expired
-        }
-        
-        guard let issuedAt = token.issuedAt,
-              abs(issuedAt.timeIntervalSince(Date.nowCoordinated)) <= issuedAtGraceInterval
-        else {
-            throw JWTError.issuedAtTimeExceedsGraceInterval
-        }
-        
-        guard token["nonce"] == context?.nonce else {
-            throw JWTError.nonceMismatch
-        }
-
-        if let maxAge = context?.maxAge {
-            guard let authTime = token.authTime else {
-                throw JWTError.invalidAuthenticationTime
-            }
-            
-            let elapsedTime = issuedAt.timeIntervalSince(authTime)
-            guard elapsedTime > 0 && elapsedTime <= maxAge else {
-                throw JWTError.exceedsMaxAge
+        for check in checks {
+            switch check {
+            case .issuer:
+                guard tokenIssuer.absoluteString == issuer.absoluteString
+                else {
+                    throw JWTError.invalidIssuer
+                }
+            case .audience:
+                guard token[.audience] == clientId else {
+                    throw JWTError.invalidAudience
+                }
+            case .scheme:
+                guard tokenIssuer.scheme == "https" else {
+                    throw JWTError.issuerRequiresHTTPS
+                }
+            case .algorithm:
+                guard token.header.algorithm == .rs256 else {
+                    throw JWTError.unsupportedAlgorithm(token.header.algorithm)
+                }
+            case .expirationTime:
+                guard let expirationTime = token.expirationTime,
+                      expirationTime > Date.nowCoordinated else {
+                    throw JWTError.expired
+                }
+            case .nonce:
+                guard token["nonce"] == context?.nonce else {
+                    throw JWTError.nonceMismatch
+                }
+            case .issuedAtTime, .authTime, .maxAge:
```

Comment:
We might want to break these out into separate cases, so they can be tested discretely.

Also, I'm not entirely certain if we need a separate "authTime" check.  There are two separate exceptions that may be thrown, but `.invalidAuthenticationTime` and `.exceedsMaxAge` are just different errors that can occur while testing `maxAge`